.github/workflows/wheels-build.yml

```diff
@@ -0,0 +1,247 @@
+name: Build release artifacts
+on:
+  workflow_call:
+    inputs:
+      default-action:
+        description: >-
+          The default action for each artifact.
+          Choose from 'build' (default) or 'skip'.
+        type: string
+        default: "build"
+        required: false
+
+      sdist:
+        description: >-
+          The action to take for the sdist.
+          Choose from 'default', 'build' or 'skip'.
+        type: string
+        default: "default"
+        required: false
+
+      wheels-tier-1:
+        description: >-
+          The action to take for Tier 1 wheels.
+          Choose from 'default', 'build' or 'skip'.
+          This builds multiple artifacts, which all match 'wheels-tier-1-*'.
+        type: string
+        default: "default"
+        required: false
+
+      wheels-32bit: 
+        description: >-
+          The action to take for Tier 1 wheels.
+          Choose from 'default', 'build' or 'skip'.
+          This builds multiple artifacts, which all match 'wheels-32bit-*'.
+        type: string
+        default: "default"
+        required: false
+
+      wheels-linux-s390x: 
+        description: >-
+          The action to take for Linux s390x wheels.
+          Choose from 'default', 'build' or 'skip'.
+        type: string
+        default: "default"
+        required: false
+
+      wheels-linux-ppc64le: 
+        description: >-
+          The action to take for Linux ppc64le wheels.
+          Choose from 'default', 'build' or 'skip'.
+        type: string
+        default: "default"
+        required: false
+
+      wheels-linux-aarch64: 
+        description: >-
+          The action to take for Linux AArch64 wheels.
+          Choose from 'default', 'build' or 'skip'.
+        type: string
+        default: "default"
+        required: false
+
+      artifact-prefix:
+        description: "A prefix to give all artifacts uploaded with 'actions/upload-artifact'."
+        type: string
+        default: ""
+        required: false
+
+      python-version:
+        description: "The Python version to use to host the build runner."
+        type: string
+        default: "3.10"
+        required: false
+
+      pgo:
+        description: "Whether to enable profile-guided optimizations for supported platforms."
+        type: boolean
+        default: true
+        required: false
+      
+
+jobs:
+  wheels-tier-1:
+    name: "Wheels / Tier 1"
+    if: (inputs.wheels-tier-1 == 'default' && inputs.default-action || inputs.wheels-tier-1) == 'build'
```

Comment:
This isn't a bash thing actually, it's GitHub Action's built-in custom expression evaluator - the `if` YAML key is part of their stuff.

`&&` and `||` are short-circuiting logical operators who use the "truthiness" to determine their result, but return the actual object.  `&&` (logical and) binds tighter than `||` (logical or), and this is pretty directly equivalent to the Python:
```python
inputs["wheels-tier-1"] == "default" and inputs["default-action"] or inputs["wheels-tier-1"]
```
The intention is to be a proper ternary operator like C's:
```c
strcmp(INPUTS_WHEELS_TIER_1, "default") == 0 ? INPUTS_DEFAULT_ACTION : INPUTS_WHEELS_TIER_1
```
(which looks more similar) or Python's
```python
inputs["default-action"] if inputs["wheels-tier-1"] == "default" else inputs["wheels-tier-1"]
```

It's not actually exactly the same, though, because if `inputs.default-action` evaluates to the empty string (which is false-y), the Boolean logic will fall through to returning `inputs.wheels-tier-1` via the logical or.  That won't happen in this case, because the default action is never supposed to be the empty string.

That said, once you're used to ternary conditionals, this pattern is relatively legible (it's a common shorthand in a lot of older languages, or very limited ones like the GitHub Actions expression DSL), and the alternatives that make it safer are typically much messier (you can make an array with the two options, then index into it with a Boolean, which coerces to integer 0 or 1, for example).